src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -417,6 +417,28 @@ case object EXTCODECOPY extends OpCode(0x3c, 4, 0, _.G_extcode) {
   }
 }
 
+case object RETURNDATASIZE extends ConstOp(0x3d)(_.returnData.size)
+
+case object RETURNDATACOPY extends OpCode(0x3e, 3, 0, _.G_verylow) {
+  protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
+    val (Seq(memOffset, dataOffset, size), stack1) = state.stack.pop(3)
+    if(dataOffset + size > state.returnData.size){
+      state.withStack(stack1).withError(ReturnDataOverflow)
```

Comment:
specification says that `if start + length overflows or results in a value larger than RETURNDATASIZE, the current call stops in an out-of-gas condition`  and when we end execution with `OutOfGas` we are setting remaining gas to 0. I am wondering if we should do the same here ?

Reply:
Yes, but we do the same thing here. In `ProgramState.toResult` method we have `if (error.exists(_.useWholeGas)) 0 else gas`. `ReturnDataOverflow` are using whole gas